Remove drawn features layer when geometries are cleared

sufyanAbbasi opened the pull request:
- Ensures that whenever the last drawn geometry is removed, regardless of whether it was deleted from the map using the draw control tools or reset using the `draw_control.reset()` method, the "Drawn Features" layer will also be removed from the layers list.

See: https://colab.research.google.com/drive/1vzHVOhtEr9lQwKqPHh3CkF2FIVcQ5SLN?usp=sharing

<img width="688" alt="Screenshot 2024-06-11 at 6 51 38 PM" src="https://github.com/gee-community/geemap/assets/12646706/38997bc4-ec74-4e43-8004-464de2aa96fd">
<img width="710" alt="Screenshot 2024-06-11 at 6 51 50 PM" src="https://github.com/gee-community/geemap/assets/12646706/b3ee2817-201e-4716-8247-591f9a037581">
<img width="690" alt="Screenshot 2024-06-11 at 6 51 59 PM" src="https://github.com/gee-community/geemap/assets/12646706/6b1dc33b-c936-4486-99d9-e0c5c38f1fa2">

giswqs reviewed (approved):
LGTM.

naschmitz commented on geemap/core.py:
```diff
@@ -15,6 +15,8 @@
 from . import map_widgets
 from . import toolbar
 
+DRAWN_FEATURES_LAYER = "Drawn Features"
```
How about `_DRAWN_FEATURES_LAYER` to mark this as a private variable?

sufyanAbbasi replied:
Done.

naschmitz reviewed (changes requested):
Could you add unit tests (as long as it doesn't take too long)?

sufyanAbbasi commented:
> Could you add unit tests (as long as it doesn't take too long)?

To make this better testable, I refactored to move the layer deletion logic to where we call `redraw_layer()` in the `handle_geometry_delete` method, and added some more implementation details to the `FakeMap`.

naschmitz reviewed (approved):
Thank you so much!
